sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -448,64 +457,74 @@ public void onActivityPrePaused(@NonNull Activity activity) {
   }
 
   @Override
-  public synchronized void onActivityPaused(final @NotNull Activity activity) {
-    // only executed if API < 29 otherwise it happens on onActivityPrePaused
-    if (!isAllActivityCallbacksAvailable) {
-      // as the SDK may gets (re-)initialized mid activity lifecycle, ensure we set the flag here as
-      // well
-      // this ensures any newly launched activity will not use the app start timestamp as txn start
-      firstActivityCreated = true;
-      if (scopes == null) {
-        lastPausedTime = AndroidDateUtils.getCurrentSentryDateTime();
-      } else {
-        lastPausedTime = scopes.getOptions().getDateProvider().now();
+  public void onActivityPaused(final @NotNull Activity activity) {
+    try (final @NotNull ISentryLifecycleToken ignored = lock.acquire()) {
+      // only executed if API < 29 otherwise it happens on onActivityPrePaused
+      if (!isAllActivityCallbacksAvailable) {
+        // as the SDK may gets (re-)initialized mid activity lifecycle, ensure we set the flag here
+        // as
+        // well
+        // this ensures any newly launched activity will not use the app start timestamp as txn
+        // start
+        firstActivityCreated = true;
+        if (scopes == null) {
+          lastPausedTime = AndroidDateUtils.getCurrentSentryDateTime();
+        } else {
+          lastPausedTime = scopes.getOptions().getDateProvider().now();
+        }
       }
     }
   }
 
   @Override
-  public synchronized void onActivityStopped(final @NotNull Activity activity) {
-    // no-op
+  public void onActivityStopped(final @NotNull Activity activity) {
+    try (final @NotNull ISentryLifecycleToken ignored = lock.acquire()) {
+      // no-op
+    }
```

Comment:
I had to take a look at each thing I refactored manually 😢 . My thinking was to make this more clear in case we replace the noop comment with some actual code in the future.

Reply:
Replaced the locking code with a comment that shall serve as a reminder